packages/go/ecosystem/ecosystem.go

```diff
@@ -0,0 +1,64 @@
+package ecosystem
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/ossf/osv-schema/packages/go/constants"
+)
+
+// Parsed represents an ecosystem-with-suffix string as defined by the spec, parsed into
+// a structured format.
+//
+// The suffix is optional and is separated from the ecosystem by a colon.
+//
+// For example, "npm:abc" would be parsed into Parsed{Ecosystem: constants.EcosystemNPM, Suffix: "abc"}
+type Parsed struct {
+	Ecosystem constants.Ecosystem
+	Suffix    string
+}
+
+// UnmarshalJSON handles unmarshalls a JSON string into a Parsed struct.
+//
+// This method implements the json.Unmarshaler interface.
+//
+//goland:noinspection GoMixedReceiverTypes
+func (p *Parsed) UnmarshalJSON(data []byte) error {
+	var str string
+	err := json.Unmarshal(data, &str)
+
+	if err != nil {
+		return err
+	}
+
+	*p = Parse(str)
+
+	return nil
+}
+
+// MarshalJSON handles marshals a Parsed struct into a JSON string.
+//
+// This method implements the json.Marshaler interface.
+//
+//goland:noinspection GoMixedReceiverTypes
+func (p Parsed) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + p.String() + `"`), nil
+}
+
+//goland:noinspection GoMixedReceiverTypes
+func (p *Parsed) String() string {
+	str := string(p.Ecosystem)
+
+	if p.Suffix != "" {
+		str += ":" + p.Suffix
+	}
+
+	return str
+}
+
+// Parse parses a string into a constants.Ecosystem and an optional suffix specified with a ":"
+func Parse(str string) Parsed {
```

Comment:
since not all ecosystems allow suffix, for those that suffix is not a thing, shall we do a check and return only the ecosystem without the suffix?

Reply:
actually maybe this is beyond what does this package do - but this still be a TODO somewhere, for example validation

Reply:
Yeah I'm of mixed feelings about that, and sadly we should decide now because we'd need to change the return to `(Parsed, error)`...

If we do want to validate it at some point, I think we should follow the `(Must)Parse` pattern, at least.

---

(I'll let others have a think and weigh in before making any changes)

Reply:
Hmm I can see the benefit of knowing Parsed contents will always be valid OSV-Schema Ecosystem, but at the same time I think that's a decent bit of logic we need to add here, and makes it less forward compatible (if you are using an older version of osv-schema golang package, you'll get errors when trying to parse new ecosystems). 

I think let's keep it "dumb" and just do a simple cut and type cast like it is currently doing.

Reply:
+1 to keeping it dumb.